/**
 * @struct HILSIMPacket
 * @brief Structure to hold data received serially from a desktop computer
 *
 * The simulated/past-launch data is streamed through serial row-by-row to TARS. TARS receives it in the HILSIM thread and populates
 * data that would otherwise be read from sensors via the received HILSIM packet. Used for rapid testing and iteration of onboard
 * hardware, GNC, and telemetry systems.
 * To generate hilsimpacket.pb.h, use the following command:
 * `python nanopb_generator/nanopb_generator.py hilsimpacket.proto`
 * To generate hilsimpacket_pb2.py, use the following command:
 * `protoc -I=. --python_out=. hilsimpacket.proto`
*/
syntax = "proto2";

message HILSIMPacket {
    // High-G IMU data
    required float imu_high_ax = 1;
    required float imu_high_ay = 2;
    required float imu_high_az = 3;

    // Barometer data
    required float barometer_altitude = 4;
    required float barometer_temperature = 5;
    required float barometer_pressure = 6;

    // Low-G IMU data
    required float imu_low_ax = 7;
    required float imu_low_ay = 8;
    required float imu_low_az = 9;

    // Low-G lsm IMU data
    required float imu_low_lsm_ax = 10;
    required float imu_low_lsm_ay = 11;
    required float imu_low_lsm_az = 12;
    required float imu_low_lsm_gx = 13;
    required float imu_low_lsm_gy = 14;
    required float imu_low_lsm_gz = 15;

    // Mag data
    required float mag_x = 16;
    required float mag_y = 17;
    required float mag_z = 18;

    // Orientation data
    required float ornt_roll = 19;
    required float ornt_pitch = 20;
    required float ornt_yaw = 21;
    required float ornt_rollv = 22;
    required float ornt_pitchv = 23;
    required float ornt_yawv = 24;
    required float ornt_rolla = 25;
    required float ornt_pitcha = 26;
    required float ornt_yawa = 27;
    required float ornt_ax = 28;
    required float ornt_ay = 29;
    required float ornt_az = 30;
    required float ornt_gx = 31;
    required float ornt_gy = 32;
    required float ornt_gz = 33;
    required float ornt_mx = 34;
    required float ornt_my = 35;
    required float ornt_mz = 36;
    required float ornt_temp = 37;
}
